test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +94,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (c.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
+        }
```

Comment:
While this does test whether mutating commands do save to the address book, it does not guarantee that non-mutating address books do not save to the address book. Perhaps an additional condition could be added to test for non-mutating commands.